### DISCUSSION TITLE
Disabling metrics and debug endpoints

### DISCUSSION THREAD

**Discussion by amalshaji-plivo** [flipt-io/flipt#2194] [Q&A]
Is there a way to disable the following endpoints?
- /debug/vars
- /debug/pprof/goroutine
- /metrics

**Comment by markphelps** [comment]
Hey @amalshaji-plivo ! great question

Not currently.. but I can def see the need for that. I have a WIP branch up to support turning off `/debug` and eventually `metrics` as well: https://github.com/flipt-io/flipt/compare/allow-disabling-profiler?expand=1

You would be able to disable `/debug` and `/metrics` via settings like:

```yaml
diagnostics:
  profiling:
    enabled: false
```

and potentially:

```yaml
diagnostics:
  metrics:
    enabled: false
```

Will need to think if `metrics` make sense under `diagnostics` or their own configuration options?

I borrowed this config from the [Grafana project](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/configure-tracing/#turn-on-profiling)

Maybe we should also think about if `profiling` should be enabled by default (as it is now) or disabled by default?

cc @GeorgeMac @yquansah

**Reply by amalshaji-plivo** [reply]
Thanks for your reply. I'm unable to access the branch.

**Reply by markphelps** [reply]
ah it was merged and deleted I think. Here is the PR: https://github.com/flipt-io/flipt/pull/2195/files

It will be available in the next release version, but if you want to try it before release, you can use the [nightly build](https://github.com/flipt-io/flipt#nightly-build)

Running the nightly image like so should disable the `/debug` endpoints. We still need to figure out the best configuration options to allow disabling `/metrics`, so that hasn't been implemented yet.

```
docker run --rm -p 8080:8080 -p 9000:9000 -t -e FLIPT_DIAGNOSTICS_PROFILING_ENABLED=false flipt/flipt:nightly
```

```
$ ~ » curl localhost:8080/debug
404 page not found
```
